text-generation-inference/server/text_generation_server/generator.py

```diff
@@ -188,7 +193,32 @@ def resume(self):
             self._mask = torch.cat([self._mask, torch.LongTensor([1])])
         self._state = Slot.State.READY
 
-    def append(self, next_token: int, next_token_text: str):
+    def _decode_next_tokens(
+        self,
+    ) -> str:
+        """Hack to hopefully support generate_stream for the maximum number of tokenizers"""
+        # We need to include the tokens that produced the last text to defeat cleanup algorithms in the decode
+        # which decide to add a space or not depending on the surrounding ids.
+        new_text = self._tokenizer.decode(self._tokens[self._next_text_token_start :], skip_special_tokens=False)
+        if new_text.endswith("�"):
```

Comment:
feels weird no?

Reply:
https://github.com/huggingface/text-generation-inference/blob/b226e469c95d5abe4a67ca0aff758b49338654b0/server/text_generation_server/models/model.py#L68C14-L68C14

Reply:
I shamelessly copied the code from the TGI implementation.